presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixConfig.java

```diff
@@ -18,22 +18,58 @@
 
 import javax.validation.constraints.NotNull;
 
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
 public class RubixConfig
 {
-    private boolean parallelWarmupEnabled = true;
+    public enum ReadMode
+    {
+        READ_THROUGH(false),
+        ASYNC(true);
+
+        private final boolean parallelWarmupEnabled;
+
+        ReadMode(boolean parallelWarmupEnabled)
+        {
+            this.parallelWarmupEnabled = parallelWarmupEnabled;
+        }
+
+        public boolean isParallelWarmupEnabled()
+        {
+            return parallelWarmupEnabled;
+        }
+
+        public static ReadMode fromString(String value)
+        {
+            switch (requireNonNull(value, "value is null").toLowerCase(ENGLISH)) {
+                case "async":
+                    return ASYNC;
+                case "read-through":
+                    return READ_THROUGH;
+            }
+
+            throw new IllegalArgumentException(format("Unrecognized value: '%s'", value));
+        }
+    }
+
     private String cacheLocation;
     private int bookKeeperServerPort = CacheConfig.DEFAULT_BOOKKEEPER_SERVER_PORT;
     private int dataTransferServerPort = CacheConfig.DEFAULT_DATA_TRANSFER_SERVER_PORT;
 
-    public boolean isParallelWarmupEnabled()
+    private ReadMode readMode = ReadMode.READ_THROUGH;
```

Comment:
We didnt discuss this much in the initial PR, we fixing on read-through mode by default now? read-through will impact the query times when query hits the cold cache

Reply:
Let's discuss on https://github.com/prestosql/presto/pull/3591.